src/Sarif.Viewer.VisualStudio/Telemetry/Events.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Sarif.Viewer.Telemetry
+{
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Contains the methods for firing telemetry events from the SARIF viewer extension.
+    /// </summary>
+    /// <remarks>
+    /// The documentation in this class is super useful for understanding where this telemetry is fired.
+    /// The documentation should also contain "why" the telemetry is being fired, i.e., why is this data needed?
+    /// Also, by having individual methods, versus calling application insights directly from the code,
+    /// it allows us to easily locate the code that is firing the telemetry.
+    /// </remarks>
+    internal class Events
+    {
+        /// <summary>
+        /// Does nothing.
+        /// </summary>
+        /// <remarks>
+        /// The reason we call track event directly here is that
+        /// we are in the middle of MEF composition\creation <see cref="TelemetryProvider.TelemetryProviderService"/>, if we called
+        /// <see cref="TelemetryProvider.TrackEvent(string)"/> that would end up being a recursive MEF composition call (and a
+        /// MEF composition exception). This method is purely used for event naming purposes.
```

Comment:
>This method is purely used for event naming purposes. [](start = 40, length = 53)

Note to self: Understand what this means.

Reply:
Now I understand, it's because you're using `[CallerMemberName]`.

---
In reply to: [505746342](https://github.com/microsoft/sarif-visualstudio-extension/pull/226#discussion_r505746342) [](ancestors = 505746342)